### ISSUE TITLE
[Report Fields] [External] Implement the policy report fields creation flow

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#43684]
[Main tracking issue].(https://github.com/Expensify/Expensify/issues/365155)

[Design doc link](https://docs.google.com/document/d/10b7pTVvgpB6_O8sRTKaCzZEVM3Ba6hxnIwyyQSSXCLY/edit). 

Implement the following section of the design doc: 
https://docs.google.com/document/d/10b7pTVvgpB6_O8sRTKaCzZEVM3Ba6hxnIwyyQSSXCLY/edit#bookmark=kix.3s2yd5r621zv

This might technically be on hold for https://github.com/Expensify/App/issues/43677 but I think Callstack can start on these in parallel using feature branch.

Include unit tests for the optimistic actions

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

**Comment by waterim** [comment]
Hello, Im Artem from Callstack and would like to help with this issue

**Comment by rezkiy37** [comment]
Hi, I’m Michael (Mykhailo) from Callstack and I would like to work on this issue.

**Comment by rezkiy37** [comment]
Actively working on the issue.

**Comment by rezkiy37** [comment]
Actively working on the issue. I've opened a draft PR. 

<details><summary>Progress</summary>
<p>

https://github.com/Expensify/App/assets/57314631/1c5955fd-902b-4e7f-8610-df78732bd2e9

</p>
</details>

**Comment by rezkiy37** [comment]
**Updates:**

1. Integrating the list values creation flow.

**Comment by rezkiy37** [comment]
**Updates:**

1. Discussion in [Slack](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1719334855161799).
2. Integrated the list values creation flow.
3. Integrated the default list value selector.

**Comment by rezkiy37** [comment]
**Updates:** 

1. Discussion in [Slack](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1719418176247909?thread_ts=1719334855.161799&cid=C06ML6X0W9L).
2. Integrated the bulk list values actions. 
3. Added all translations.
4. Ingteragrated validations.
5. Integrated a draft create function.

**Comment by rezkiy37** [comment]
**Updates:**

1. Discussion in [Slack](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1719418176247909?thread_ts=1719334855.161799&cid=C06ML6X0W9L).
2. Applied review suggestions.
3. Integrated `createReportField` function.
4. Tested how it works with API.
5. Started creating automated tests.

**Comment by rezkiy37** [comment]
I've opened the PR (https://github.com/Expensify/App/pull/44275) for review.

**Comment by mountiny** [comment]
I believe the PR was merged so we are done here
